src/test/java/seedu/address/model/food/FoodTest.java

```diff
@@ -1,4 +1,74 @@
 package seedu.address.model.food;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import static seedu.address.logic.commands.CommandTestUtil.VALID_NAME_BOB;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_HUSBAND;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalFood.PASTA;
+import static seedu.address.testutil.TypicalFood.BURGER;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.FoodBuilder;
+import seedu.address.testutil.PersonBuilder;
+
 public class FoodTest {
+
+    @Test
+    public void asObservableList_modifyList_throwsUnsupportedOperationException() {
+        Food food = new FoodBuilder().build();
+        assertThrows(UnsupportedOperationException.class, () -> food.getTags().remove(0));
+    }
+
+    @Test
+    public void isSameFood() {
+        // same object -> returns true
+        assertTrue(PASTA.isSameFood(PASTA));
+
+        // null -> returns false
+        assertFalse(PASTA.isSameFood(null));
+
+        // different name -> returns false
+        Food editedAlice = new FoodBuilder(PASTA).withName(VALID_NAME_BOB).build();
+        assertFalse(PASTA.isSameFood(editedAlice));
+
+        // same name, same phone, different attributes -> returns true
+        editedAlice = new FoodBuilder(PASTA)
+                .withTags(VALID_TAG_HUSBAND).build();
+        assertTrue(PASTA.isSameFood(editedAlice));
+
+        // same name, same phone, same email, different attributes -> returns true
+        editedAlice = new FoodBuilder(PASTA).withTags(VALID_TAG_HUSBAND).build();
+        assertTrue(PASTA.isSameFood(editedAlice));
+    }
```

Comment:
With regards to the comments, I think you may need to edit the comments as it seems to reflect a non-attributes of the Food Class in "same phone" and "same email".